### ISSUE TITLE
unused-fields - Too aggressive encouraging bad practices to circumvent the warning.

### ISSUE THREAD

**Issue by shanshrew** [relayjs/eslint-plugin-relay#164]
I believe `unused-fields` is configured too aggressively. For example this.

```
const user = useFragment(... some fields)
const permisions = usePermissionSet(user);
```

`unused-fields` will flag on all fields in that query, even though they could all be used in that other file, it just doesn't know. If it doesn't know it shouldn't flag everything it should flag nothing. 

Iv'e reviewed a lot of PR's where people are basically just inlining everything into the component not because it's good component design or architecture but purely to silence the linter, because disabling it for every field feels wrong.

`usePermissionsSet` can be a hook, or just a reusable function that consumes the data in any way.

Is there any chance we can look at having this changed; or a {strict} option? or {loose} option?

**Comment by captbaritone** [comment]
This is by design. If a helper hook is using server data, it should define its own fragment describing the data it needs. Generally if a helper function/hook is using data and that data is expected to be in the same shape as server data, it should have a fragment. If the helper function is not tied to the shape of the data on the server, then it should accept individual scalar fields and would thus not be flagged by the lint rule.

You can read more [here](https://relay.dev/blog/2023/10/24/how-relay-enables-optimal-data-fetching/) about how this helps avoid the failure mode where it's unclear when a field can safely be removed from a fragment/query.
